### DISCUSSION TITLE
WSL start error

### DISCUSSION THREAD

**Discussion by j7soft** [microsoft/vscode-discussions#1352] [Extension Development QnA] [answered]
Hi! i have got the error while i start VS Code in WSL terminal by typing the command "code ." 
What is the reason? 

TERMINAL output is as below:

[2024-08-05 02:26:45.104] Download in background is enabled
[2024-08-05 02:26:45.391] Using executable (2) to forward requests from local 6587 to remote 46833
[2024-08-05 02:26:45.392] NodeExecServer run: C:\WINDOWS\System32\wsl.exe -d Ubuntu-24.04 -e /home/jsoft/.vscode-server/bin/b1c0a14de1414fcdaa400695b4db1c0799bc3124/node -e const net = require('net'); process.stdin.pause(); const client = net.createConnection({ host: '127.0.0.1', port: 46833 }, () => { client.pipe(process.stdout); process.stdin.pipe(client); }); client.on('close', function (hadError) { console.error(hadError ? 'Remote close with error' : 'Remote close'); process.exit(hadError ? 1 : 0); }); client.on('error', function (err) { process.stderr.write(err && (err.stack || err.message) || String(err)); });
[2024-08-05 02:26:50.630] Exec (2) stderr: Remote close
[2024-08-05 02:26:50.671] Exec (2) has terminated  with status 0 and message undefined.
[2024-08-05 02:26:50.797] Exec (2): Local close

I am getting the Log at the server side as below
jsoft@DESKTOP-3QAL0OP:~/wsl-book/chapter-09/web-app$ VSCODE_WSL_DEBUG_INFO=true c
ode .
+ COMMIT=b1c0a14de1414fcdaa400695b4db1c0799bc3124
+ APP_NAME=code
+ QUALITY=stable
+ NAME=Code
+ SERVERDATAFOLDER=.vscode-server
+ realpath /mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/bin/code
+ dirname /mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/bin/code
+ dirname /mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/bin
+ VSCODE_PATH=/mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code
+ ELECTRON=/mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/Code.exe
+ IN_WSL=false
+ [ -n Ubuntu-24.04 ]
+ IN_WSL=true
+ [ true = true ]
+ export WSLENV=ELECTRON_RUN_AS_NODE/w:WT_SESSION:WT_PROFILE_ID:
+ wslpath -m /mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js
+ CLI=C:/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js
+ WSL_EXT_ID=ms-vscode-remote.remote-wsl
+ ELECTRON_RUN_AS_NODE=1 /mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/Code.exe C:/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/resources/app/out/cli.js --locate-extension ms-vscode-remote.remote-wsl
+ cat /tmp/remote-wsl-loc.txt
+ WSL_EXT_WLOC=c:\Users\jsoft\.vscode\extensions\ms-vscode-remote.remote-wsl-0.88.2
+ [ -n c:\Users\jsoft\.vscode\extensions\ms-vscode-remote.remote-wsl-0.88.2 ]
+ wslpath -u c:\Users\jsoft\.vscode\extensions\ms-vscode-remote.remote-wsl-0.88.2
+ WSL_CODE=/mnt/c/Users/jsoft/.vscode/extensions/ms-vscode-remote.remote-wsl-0.88.2/scripts/wslCode.sh
+ /mnt/c/Users/jsoft/.vscode/extensions/ms-vscode-remote.remote-wsl-0.88.2/scripts/wslCode.sh b1c0a14de1414fcdaa400695b4db1c0799bc3124 stable /mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/Code.exe code .vscode-server .
+ [ -z .vscode-server ]
+ [ .vscode-server = .vscode-insiders ]
+ [ ! -t 0 ]
+ VSCODE_REMOTE_BIN=/home/jsoft/.vscode-server/bin
+ AUTHORITY=wsl+default
+ [ Ubuntu-24.04 ]
+ AUTHORITY=wsl+Ubuntu-24.04
+ dirname /mnt/c/Users/jsoft/.vscode/extensions/ms-vscode-remote.remote-wsl-0.88.2/scripts/wslCode.sh
+ /mnt/c/Users/jsoft/.vscode/extensions/ms-vscode-remote.remote-wsl-0.88.2/scripts/wslDownload.sh b1c0a14de1414fcdaa400695b4db1c0799bc3124 stable /home/jsoft/.vscode-server/bin
+ [ ! -d /home/jsoft/.vscode-server/bin/b1c0a14de1414fcdaa400695b4db1c0799bc3124 ]
+ RC=0
+ [ 0 -eq 100 ]
+ [ 0 -ne 0 ]
+ mktemp /tmp/vscode-distro-env.XXXXXX
+ STORED_ENV=/tmp/vscode-distro-env.AOE7lS
+ env
+ dirname /mnt/c/Users/jsoft/.vscode/extensions/ms-vscode-remote.remote-wsl-0.88.2/scripts/wslCode.sh
+ VSCODE_CLIENT_COMMAND=/mnt/c/Users/jsoft/AppData/Local/Programs/Microsoft VS Code/Code.exe VSCODE_CLIENT_COMMAND_CWD=/mnt/c/Users/jsoft/.vscode/extensions/ms-vscode-remote.remote-wsl-0.88.2/scripts VSCODE_CLI_AUTHORITY=wsl+Ubuntu-24.04 VSCODE_CLI_REMOTE_ENV=/tmp/vscode-distro-env.AOE7lS VSCODE_STDIN_FILE_PATH= WSLENV=VSCODE_CLI_REMOTE_ENV/w:ELECTRON_RUN_AS_NODE/w:WT_SESSION:WT_PROFILE_ID: /home/jsoft/.vscode-server/bin/b1c0a14de1414fcdaa400695b4db1c0799bc3124/bin/remote-cli/code .
+ exit 0

**Comment by isidorn** [comment, marked as answer]
Looks like you are a VS Code user, and this discussion is for VS Code extension authors.
Please ask this on stack overflow.

**Reply by j7soft** [reply]
thank you. I resolved the problem myself. It was because of some configuration error.
